pkg/controller/node/node_controller.go

```diff
@@ -763,28 +785,74 @@ func (ctrl *Controller) handleErr(err error, key interface{}) {
 // 2. If a MachineConfig changes, we should wait for the OS image build to be
 // ready so we can update both the nodes' desired MachineConfig and desired
 // image annotations simultaneously.
+
+func (ctrl *Controller) GetConfigAndBuild(pool *mcfgv1.MachineConfigPool) (*mcfgv1alpha1.MachineOSConfig, *mcfgv1alpha1.MachineOSBuild, error) {
+	var ourConfig *mcfgv1alpha1.MachineOSConfig
+	var ourBuild *mcfgv1alpha1.MachineOSBuild
+	configList, err := ctrl.moscLister.List(labels.Everything())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, config := range configList {
+		if config.Spec.MachineConfigPool.Name == pool.Name {
```

Comment:
thought: I wish this was more ergonomic. I know that listers support label selectors, but then we'd need something to set a label on each MachineOSConfig and MachineOSBuild object. It'd be nice if listers supported field selectors too.

(To be clear: I'm not asking you to add labels.)

Reply:
agree, I think for GA we should support matching labels on these new objects